core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2194,7 +2193,14 @@ private Expression maybeBox(Expression expression) {
     @Override Expression implementSafe(RexToLixTranslator translator,
         RexCall call, List<Expression> argValueList) {
       final Expression argValue = argValueList.get(0);
-      final UnaryExpression e = Expressions.makeUnary(expressionType, argValue);
+
+      final Expression e;
+      if (expressionType == ExpressionType.Negate && argValue.type == BigDecimal.class) {
+        e = Expressions.call(argValue, "negate");
```

Comment:
replace "negate" string here with a reference to a method in BuiltInMethod

Reply:
OK